### ISSUE TITLE
change of init_weight not functioning in DeepLabCut3.0

### ISSUE THREAD

**Issue by lijiejue-grad** [DeepLabCut/DeepLabCut#2722]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

I was aiming to continue to train a network following a best snapshot previously trained in another shuffle. But no matter how I tried to change the param **init_weight** (in pose_cfg.yaml or directly set init_weight='path to my snapshot without extension in name' when using the deeplabcut.train_network function), the Pytorch seems to always stick to loading pretrained weight downloaded from Hugging Face.

I was wondering whether it is just a network training function not yet implemented in DeepLabCut3.0 or there's anything I mistake in use. Any comments could help! 

Thank you for building DLC in the Pytorch platform. It really speeds up my training and gets better performance. Wishing to see more great DLC features and documentations completed in this new awesome DeepLabCut3 version.

### Operating System

OS: ubuntu 22.04

### DeepLabCut version

dlc version: 3.0.0rc3

### DeepLabCut mode

single animal

### Device type

gpu: NVIDIA GeForce RTX 4090

### Steps To Reproduce

conda activate deeplabcut3
python
import deeplabcut
deeplabcut.train_network('/media/lijiejue/Elements/Deeplabcut_projects_and_Anipose_projects/20231225-test5-test6-LRimprinted-LJJ-2024-08-20/config.yaml',shuffle=2,epochs=2000,init_weight='/media/lijiejue/Elements/Deeplabcut_projects_and_Anipose_projects/20231225-test5-test6-LRimprinted-LJJ-2024-08-20/dlc-models-pytorch/iteration-0/20231225-test5-test6-LRimprintedAug20-trainset85shuffle1/train/snapshot-850',keepdeconvweights=True)

### Relevant log output

```shell
2024-08-21 16:32:44 Training with configuration:
2024-08-21 16:32:44 data:
2024-08-21 16:32:44   colormode: RGB
2024-08-21 16:32:44   inference:
2024-08-21 16:32:44     normalize_images: True
2024-08-21 16:32:44   train:
2024-08-21 16:32:44     affine:
2024-08-21 16:32:44       p: 0.5
2024-08-21 16:32:44       rotation: 30
2024-08-21 16:32:44       scaling: [1.0, 1.0]
2024-08-21 16:32:44       translation: 0
2024-08-21 16:32:44     collate:
2024-08-21 16:32:44       type: ResizeFromDataSizeCollate
2024-08-21 16:32:44       min_scale: 0.4
2024-08-21 16:32:44       max_scale: 1.0
2024-08-21 16:32:44       min_short_side: 128
2024-08-21 16:32:44       max_short_side: 3500
2024-08-21 16:32:44       max_size: 3500
2024-08-21 16:32:44       multiple_of: 32
2024-08-21 16:32:44       to_square: False
2024-08-21 16:32:44     covering: False
2024-08-21 16:32:44     gaussian_noise: 12.75
2024-08-21 16:32:44     hist_eq: False
2024-08-21 16:32:44     motion_blur: False
2024-08-21 16:32:44     normalize_images: True
2024-08-21 16:32:44 device: cuda:0
2024-08-21 16:32:44 metadata:
2024-08-21 16:32:44   project_path: /media/lijiejue/Elements/Deeplabcut_projects_and_Anipose_projects/20231225-test5-test6-LRimprinted-LJJ-2024-08-20
2024-08-21 16:32:44   pose_config_path: /media/lijiejue/Elements/Deeplabcut_projects_and_Anipose_projects/20231225-test5-test6-LRimprinted-LJJ-2024-08-20/dlc-models-pytorch/iteration-0/20231225-test5-test6-LRimprintedAug20-trainset85shuffle2/train/pose_cfg.yaml
2024-08-21 16:32:44   bodyparts: ['EarL', 'EarR', 'Snout', 'SpineF', 'SpineM', 'Tail(base)', 'Tail(mid)', 'Tail(end)', 'ForepawL', 'WristL', 'ElbowL', 'ShoulderL', 'ForepawR', 'WristR', 'ElbowR', 'ShoulderR', 'HindpawL', 'AnkleL', 'KneeL', 'HindpawR', 'AnkleR', 'KneeR']
2024-08-21 16:32:44   unique_bodyparts: []
2024-08-21 16:32:44   individuals: ['animal']
2024-08-21 16:32:44   with_identity: None
2024-08-21 16:32:44 method: bu
2024-08-21 16:32:44 model:
2024-08-21 16:32:44   backbone:
2024-08-21 16:32:44     type: ResNet
2024-08-21 16:32:44     model_name: resnet50_gn
2024-08-21 16:32:44     output_stride: 16
2024-08-21 16:32:44     freeze_bn_stats: False
2024-08-21 16:32:44     freeze_bn_weights: False
2024-08-21 16:32:44   backbone_output_channels: 2048
2024-08-21 16:32:44   heads:
2024-08-21 16:32:44     bodypart:
2024-08-21 16:32:44       type: HeatmapHead
2024-08-21 16:32:44       weight_init: normal
2024-08-21 16:32:44       predictor:
2024-08-21 16:32:44         type: HeatmapPredictor
2024-08-21 16:32:44         apply_sigmoid: False
2024-08-21 16:32:44         clip_scores: True
2024-08-21 16:32:44         location_refinement: True
2024-08-21 16:32:44         locref_std: 7.2801
2024-08-21 16:32:44       target_generator:
2024-08-21 16:32:44         type: HeatmapGaussianGenerator
2024-08-21 16:32:44         num_heatmaps: 22
2024-08-21 16:32:44         pos_dist_thresh: 17
2024-08-21 16:32:44         heatmap_mode: KEYPOINT
2024-08-21 16:32:44         generate_locref: True
2024-08-21 16:32:44         locref_std: 7.2801
2024-08-21 16:32:44       criterion:
2024-08-21 16:32:44         heatmap:
2024-08-21 16:32:44           type: WeightedMSECriterion
2024-08-21 16:32:44           weight: 1.0
2024-08-21 16:32:44         locref:
2024-08-21 16:32:44           type: WeightedHuberCriterion
2024-08-21 16:32:44           weight: 0.05
2024-08-21 16:32:44       heatmap_config:
2024-08-21 16:32:44         channels: [2048, 22]
2024-08-21 16:32:44         kernel_size: [3]
2024-08-21 16:32:44         strides: [2]
2024-08-21 16:32:44       locref_config:
2024-08-21 16:32:44         channels: [2048, 44]
2024-08-21 16:32:44         kernel_size: [3]
2024-08-21 16:32:44         strides: [2]
2024-08-21 16:32:44 net_type: resnet_50
2024-08-21 16:32:44 runner:
2024-08-21 16:32:44   type: PoseTrainingRunner
2024-08-21 16:32:44   gpus: None
2024-08-21 16:32:44   key_metric: test.mAP
2024-08-21 16:32:44   key_metric_asc: True
2024-08-21 16:32:44   eval_interval: 50
2024-08-21 16:32:44   optimizer:
2024-08-21 16:32:44     type: AdamW
2024-08-21 16:32:44     params:
2024-08-21 16:32:44       lr: 0.0001
2024-08-21 16:32:44   scheduler:
2024-08-21 16:32:44     type: LRListScheduler
2024-08-21 16:32:44     params:
2024-08-21 16:32:44       lr_list: [[1e-05], [1e-06]]
2024-08-21 16:32:44       milestones: [1800, 1900]
2024-08-21 16:32:44   snapshots:
2024-08-21 16:32:44     max_snapshots: 40
2024-08-21 16:32:44     save_epochs: 50
2024-08-21 16:32:44     save_optimizer_state: False
2024-08-21 16:32:44 train_settings:
2024-08-21 16:32:44   batch_size: 1
2024-08-21 16:32:44   dataloader_workers: 50
2024-08-21 16:32:44   dataloader_pin_memory: True
2024-08-21 16:32:44   display_iters: 1000
2024-08-21 16:32:44   epochs: 2000
2024-08-21 16:32:44   seed: 42




2024-08-21 16:32:44 init_weight: /media/lijiejue/Elements/Deeplabcut_projects_and_Anipose_projects/20231225-test5-test6-LRimprinted-LJJ-2024-08-20/dlc-models-pytorch/iteration-0/20231225-test5-test6-LRimprintedAug20-trainset85shuffle1/train/snapshot-850
2024-08-21 16:32:44 Loading pretrained weights from Hugging Face hub (timm/resnet50_gn.a1h_in1k)
2024-08-21 16:32:44 [timm/resnet50_gn.a1h_in1k] Safe alternative available for 'pytorch_model.bin' (as 'model.safetensors'). Loading weights using safetensors.







2024-08-21 16:32:44 Data Transforms:
2024-08-21 16:32:44   Training:   Compose([
  Affine(always_apply=False, p=0.5, interpolation=1, mask_interpolation=0, cval=0, mode=0, scale={'x': (1.0, 1.0), 'y': (1.0, 1.0)}, translate_percent=None, translate_px={'x': (0, 0), 'y': (0, 0)}, rotate=(-30, 30), fit_output=False, shear={'x': (0.0, 0.0), 'y': (0.0, 0.0)}, cval_mask=0, keep_ratio=True, rotate_method='largest_box'),
  GaussNoise(always_apply=False, p=0.5, var_limit=(0, 162.5625), per_channel=True, mean=0),
  Normalize(always_apply=False, p=1.0, mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225], max_pixel_value=255.0),
], p=1.0, bbox_params={'format': 'coco', 'label_fields': ['bbox_labels'], 'min_area': 0.0, 'min_visibility': 0.0, 'min_width': 0.0, 'min_height': 0.0, 'check_each_transform': True}, keypoint_params={'format': 'xy', 'label_fields': ['class_labels'], 'remove_invisible': False, 'angle_in_degrees': True, 'check_each_transform': True}, additional_targets={}, is_check_shapes=True)
2024-08-21 16:32:44   Validation: Compose([
  Normalize(always_apply=False, p=1.0, mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225], max_pixel_value=255.0),
], p=1.0, bbox_params={'format': 'coco', 'label_fields': ['bbox_labels'], 'min_area': 0.0, 'min_visibility': 0.0, 'min_width': 0.0, 'min_height': 0.0, 'check_each_transform': True}, keypoint_params={'format': 'xy', 'label_fields': ['class_labels'], 'remove_invisible': False, 'angle_in_degrees': True, 'check_each_transform': True}, additional_targets={}, is_check_shapes=True)
2024-08-21 16:32:45 Using custom collate function: {'type': 'ResizeFromDataSizeCollate', 'min_scale': 0.4, 'max_scale': 1.0, 'min_short_side': 128, 'max_short_side': 3500, 'max_size': 3500, 'multiple_of': 32, 'to_square': False}
2024-08-21 16:32:45 
Note: According to your model configuration, you're training with batch size 1 and/or ``freeze_bn_stats=false``. This is not an optimal setting if you have powerful GPUs.
This is good for small batch sizes (e.g., when training on a CPU), where you should keep ``freeze_bn_stats=true``.
If you're using a GPU to train, you can obtain faster performance by setting a larger batch size (the biggest power of 2 where you don't geta CUDA out-of-memory error, such as 8, 16, 32 or 64 depending on the model, size of your images, and GPU memory) and ``freeze_bn_stats=false`` for the backbone of your model. 
This also allows you to increase the learning rate (empirically you can scale the learning rate by sqrt(batch_size) times).

2024-08-21 16:32:45 Using 289 images and 51 for testing
2024-08-21 16:32:45 
Starting pose model training...
--------------------------------------------------
2024-08-21 16:33:36 Epoch 1/2000 (lr=0.0001), train loss 0.01184
2024-08-21 16:34:28 Epoch 2/2000 (lr=0.0001), train loss 0.00914
2024-08-21 16:35:19 Epoch 3/2000 (lr=0.0001), train loss 0.00852
2024-08-21 16:36:11 Epoch 4/2000 (lr=0.0001), train loss 0.00820
2024-08-21 16:37:03 Epoch 5/2000 (lr=0.0001), train loss 0.00789
2024-08-21 16:37:56 Epoch 6/2000 (lr=0.0001), train loss 0.00766
2024-08-21 16:38:48 Epoch 7/2000 (lr=0.0001), train loss 0.00742
2024-08-21 16:39:40 Epoch 8/2000 (lr=0.0001), train loss 0.00724
2024-08-21 16:40:31 Epoch 9/2000 (lr=0.0001), train loss 0.00709
etc...
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by lijiejue-grad** [comment]
I've solved this problem by looking for the train_network function codes.
The correct command should be:

```python
deeplabcut.pose_estimation_pytorch.apis.train.train_network(
    '/media/lijiejue/Elements/Deeplabcut_projects_and_Anipose_projects/20231225-test5-test6-LRimprinted-LJJ-2024-08-20/config.yaml',
    shuffle=3,
    snapshot_path='/media/lijiejue/Elements/Deeplabcut_projects_and_Anipose_projects/20231225-test5-test6-LRimprinted-LJJ-2024-08-20/dlc-models-pytorch/iteration-0/20231225-test5-test6-LRimprintedAug20-trainset85shuffle2/train/snapshot-1850.pt',
)
```

**Comment by wheelern** [comment]
Is this documented anywhere? I had the same problem. A search for `snapshot_path` in the docs returns no results.

**Comment by n-poulsen** [comment]
The most up-to-date API for the PyTorch engine is always documented directly in the code through docstrings (e.g., you can see the docs for `train_network` [here](https://github.com/DeepLabCut/DeepLabCut/blob/78cc44eaea296fc2a6f8979027497298348359aa/deeplabcut/pose_estimation_pytorch/apis/train.py#L191)). All high-level API methods (such as `train_network`, `evaluate_network`, `analyze_videos`, ...) are implemented and documented in the [`deeplabcut/pose_estimation_pytorch/apis`](https://github.com/DeepLabCut/DeepLabCut/tree/pytorch_dlc/deeplabcut/pose_estimation_pytorch/apis) folder.

We'll work on integrating the new parameters for the PyTorch API in our docs online, but for now looking at `docstrings` is your best option. You can also get those docs through `ipython`:

![Screenshot 2024-09-09 at 09 50 36](https://github.com/user-attachments/assets/4122afdb-2907-4f64-a5ad-25752138d9b7)
